### PR TITLE
Upgrade to NET 9.0.

### LINKED ISSUE #20561: Upgrade to .NET 9.0 RC.1
Once it is released, we will upgrade to .NET 9.0 RC.1.
ABP 9.0-RC.1 will be based on .NET 9.0-RC.x, and final ABP 9.0 will be based on .NET 9.0.

---

https://dotnet.microsoft.com/en-us/download/dotnet/9.0

https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-9/overview

https://learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-9.0

https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-9.0/whatsnew

### PR DESCRIPTION
**Pull request by maliming**
Resolves #20561

---

The changes to upgrade net 8 to net 9:

Install the `9.0-rc1` SDK: https://dotnet.microsoft.com/en-us/download/dotnet/9.0
Upgrade `dotnet-ef` tool version to `9.0.0-rc.1.24451.1`.
`dotnet tool install --global dotnet-ef --version 8.0.8`

1. Change all `TargetFramework` from `net8.0` to `net9.0`.
2. Upgrade all Microsoft nuget packages to `9.0.0-rc.1.xxx`.
3. Update the `version` to `9.0.100-rc.1.24452.12` in `global.json` if you have this file.
4. Change `app.UseStaticFiles()` to `app.MapAbpStaticAssets()`. **The method name** is `MapAbpStaticAssets` instead of `MapStaticAssets`.

---

#### About `MapAbpStaticAssets`

Some `JavaScript/CSS/Images` files exist in the [Virtual File System](https://abp.io/docs/latest/framework/infrastructure/virtual-file-system?_redirected=B8ABF606AA1BDF5C629883DF1061649A), but ASP NET Core 9's `MapStaticAssets` can't handle them.

We need additional `StaticFileMiddleware` to serve these files. This is why we need to call `MapAbpStaticAssets`.

The `MapAbpStaticAssets` will add some `StaticFileMiddleware` to make sure virtual files work.

The source code of `MapAbpStaticAssets`: https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.AspNetCore/Microsoft/AspNetCore/Builder/AbpApplicationBuilderExtensions.cs#L129-L198

See [Static files in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/static-files?view=aspnetcore-9.0) to learn about the new `MapStaticAssets` feature.

---

#### About `IAbpHybirdCache<,>`

I have added the new `HybirdCache` in a [separate PR](https://github.com/abpframework/abp/pull/20859). 

**Attention**: ASP.NET Core HybirdCache will use memory cache as `L1` cache. When your application has multiple instances, if instance **A** deletes a cache key, instance **B**'s `L1` cache remains valid. To solve this case, you may need to use [Distributed Event Bus](https://abp.io/docs/latest/framework/infrastructure/event-bus/distributed).

See [HybridCache library in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/performance/caching/hybrid?view=aspnetcore-9.0&viewFallbackFrom=aspnetcore-8.0) to learn about the new `HybridCache ` feature.

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -51,69 +51,69 @@
...
+    <PackageVersion Include="Microsoft.IdentityModel.Protocols.OpenIdConnect" Version="8.0.2" />
+    <PackageVersion Include="Microsoft.IdentityModel.Tokens" Version="8.0.2" />
+    <PackageVersion Include="Microsoft.IdentityModel.JsonWebTokens" Version="8.0.2" />
     <PackageVersion Include="Minio" Version="6.0.2" />
     <PackageVersion Include="MongoDB.Driver" Version="2.22.0" />
     <PackageVersion Include="NEST" Version="7.17.5" />
     <PackageVersion Include="Newtonsoft.Json" Version="13.0.3" />
     <PackageVersion Include="Nito.AsyncEx.Context" Version="5.1.2" />
-    <PackageVersion Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="8.0.2" />
+    <PackageVersion Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="9.0.0-preview.7" />
```
Comment (salihozkara) [inline Directory.Packages.props]:
```suggestion
    <PackageVersion Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="9.0.0-rc.1" />
```

Comment (salihozkara) [reply Directory.Packages.props]: https://www.nuget.org/packages/Npgsql.EntityFrameworkCore.PostgreSQL/9.0.0-rc.1#versions-body-tab

Comment (ebicoglu) [reply Directory.Packages.props]: PostgreSQL supports .NET9 https://github.com/npgsql/efcore.pg/blob/main/test/Directory.Build.props#L6

Comment (maliming) [reply Directory.Packages.props]: https://github.com/abpframework/abp/pull/20803/commits/0feddd835c6a2cebe61b321c7707208719c05559

**Comment by skoc10** [comment]
I have tested nightly build pipeline with this branch, there is no any error.

**Comment by MarkCiliaVincenti** [comment]
Is there a reason why you are not multi-targeting these libraries? I can see some people preferring to use .NET 8.0 which is LTS. I would expect ABP to fully support an LTS version of .NET.

**Review by realLiangshiwei — APPROVED** [review]

**Review by MarkCiliaVincenti — APPROVED** [review]

**Comment by MarkCiliaVincenti** [comment]
Is it on purpose that you're also targeting frameworks which support thread aborts and which could lead to issues? Is there a reason why someone would want to use abp on anything before net5?

**Review by enisn — APPROVED** [review]

**Pull request merged**